### PR TITLE
MINIFICPP-1383 fix intdiv_ceil

### PR DESCRIPTION
**Pull request by szaszm**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by lordgamez — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

File: libminifi/test/unit/GeneralUtilsTest.cpp [hunk 1]
```diff
@@ -32,18 +32,20 @@ TEST_CASE("GeneralUtils::make_unique", "[make_unique]") {
...
-  REQUIRE(-1 == utils::intdiv_ceil(-3, 3));
-  REQUIRE(-1 == utils::intdiv_ceil(-4, 3));
-}
+static_assert(0 == utils::intdiv_ceil(0, 1), "");
+static_assert(0 == utils::intdiv_ceil(0, 2), "");
+static_assert(1 == utils::intdiv_ceil(1, 2), "");
+static_assert(1 == utils::intdiv_ceil(1, 3), "");
+static_assert(1 == utils::intdiv_ceil(3, 3), "");
+static_assert(2 == utils::intdiv_ceil(4, 3), "");
+static_assert(2 == utils::intdiv_ceil(4, 3), "");
```
Comment (fgerlits) [inline libminifi/test/unit/GeneralUtilsTest.cpp:41]: very minor (and old code), but you have the same (4, 3) test twice

File: libminifi/test/unit/GeneralUtilsTest.cpp [hunk 2]
```diff
@@ -32,18 +32,20 @@ TEST_CASE("GeneralUtils::make_unique", "[make_unique]") {
...
+static_assert(0 == utils::intdiv_ceil(-1, 3), "");
+static_assert(-1 == utils::intdiv_ceil(-3, 3), "");
+static_assert(-1 == utils::intdiv_ceil(-4, 3), "");
+static_assert(2 == utils::intdiv_ceil(-4, -3), "");
+static_assert(2 == utils::intdiv_ceil(-5, -3), "");
+static_assert(0 == utils::intdiv_ceil(1, -3), "");
+static_assert(-1 == utils::intdiv_ceil(5, -3), "");
```
Comment (fgerlits) [inline libminifi/test/unit/GeneralUtilsTest.cpp:48]: you could add some more tests:
 * (0, negative number)
 * evenly divisible but not equal, eg. (6, 2) and variations: (-6, 2), (-6, -2), (6, -2)
 * test that division by 0 doesn't compile

Comment (szaszm) [reply libminifi/test/unit/GeneralUtilsTest.cpp:48]: Not sure how can I test that 0 doesn't compile. Will do the rest.

File: libminifi/test/unit/GeneralUtilsTest.cpp (hunk 1 above)
Comment (szaszm) [reply libminifi/test/unit/GeneralUtilsTest.cpp]: fixed in 585e874

File: libminifi/test/unit/GeneralUtilsTest.cpp (hunk 2 above)
Comment (szaszm) [reply libminifi/test/unit/GeneralUtilsTest.cpp:49]: done in 585e874

Comment (adamdebreceni) [reply libminifi/test/unit/GeneralUtilsTest.cpp:49]: you could SFINAE it:

```
template<int n, int d, typename = void>
struct does_compile : std::false_type {};

template<int n, int d>
struct does_compile<n, d, decltype(intdiv_ceil(n, d))> : std::true_type {};
```

Comment (adamdebreceni) [reply libminifi/test/unit/GeneralUtilsTest.cpp:49]:
```

template<int n, int d, typename = void>
struct does_compile : std::false_type {};

template<int n, int d>
struct does_compile<n, d, typename std::enable_if<(intdiv_ceil(n, d), true)>::type> : std::true_type {};
```

**Review by fgerlits — APPROVED** [review]

File: libminifi/test/unit/GeneralUtilsTest.cpp (hunk 2 above)
Comment (szaszm) [reply libminifi/test/unit/GeneralUtilsTest.cpp:49]: I hope you don't mind me setting you as the author as you basically submitted the whole code in b11f101

Comment (adamdebreceni) [reply libminifi/test/unit/GeneralUtilsTest.cpp:49]: I'll survive somehow :) a slight addition would be to include a case that should compile so we could check for false negatives, e.g. if by accident `does_compile<...>::value` always returns `false`, basically a test for the test

Comment (szaszm) [reply libminifi/test/unit/GeneralUtilsTest.cpp:49]: done in c26cf15

**Review by adamdebreceni — APPROVED** [review]

**Pull request closed without merging**